test/extensions/filters/http/ext_proc/ext_proc_integration_test.cc

```diff
@@ -741,24 +748,60 @@ class ExtProcIntegrationTest : public HttpIntegrationTest,
     return response;
   }
 
-  void serverReceiveHeaderDuplexStreamed(ProcessingRequest& header_request) {
-    EXPECT_TRUE(grpc_upstreams_[0]->waitForHttpConnection(*dispatcher_, processor_connection_));
-    EXPECT_TRUE(processor_connection_->waitForNewStream(*dispatcher_, processor_stream_));
-    EXPECT_TRUE(processor_stream_->waitForGrpcMessage(*dispatcher_, header_request));
-    EXPECT_TRUE(header_request.has_request_headers());
+  void serverReceiveHeaderDuplexStreamed(ProcessingRequest& header, bool first_message = true,
+                                         bool response = false) {
+    if (first_message) {
+      EXPECT_TRUE(grpc_upstreams_[0]->waitForHttpConnection(*dispatcher_, processor_connection_));
```

Comment:
A pattern I use sometimes when looking at expectations in a helper function is to provide more context by doing `<< something` in the EXPECT call. That way, if you have multiple calls to it from the same test method, and one of them fails, you'll know which one. You might pass a string for `something` into the function, or compute `something` from the args.

Reply:
For waitForHttpConnection() and waitForNewStream(), these functions are already returning those "something", and printed out in the log, for example: https://github.com/envoyproxy/envoy/blob/66cc2175fe5044117c9f00af8d09293012778000/test/integration/fake_upstream.cc#L798.

I added << "something in the below two places.